apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/signature/SignatureParser.java

```diff
@@ -128,38 +130,62 @@ private void parse(String query, StringBuilder signature) {
             case UPDATE:
                 signature.append("UPDATE");
                 // Scan for the table name
-                boolean hasPeriod = false, hasFirstPeriod = false;
+                boolean hasPeriod = false, hasFirstPeriod = false, inQuotes = false, inDbLink = false;
+        		StringBuilder targetBuilder=signature;
                 if (scanner.scanToken(IDENT)) {
                     signature.append(' ');
                     scanner.appendCurrentTokenText(signature);
-                    for (Scanner.Token t = scanner.scan(); t != EOF; t = scanner.scan()) {
+                    for (Scanner.Token t = scanner.scan(false); t != EOF; t = scanner.scan(false)) {
```

Comment:
I think we could make a couple of simplifications:

Don't tokenise the dblink string, reverting the additional parameter to `scan`, and just assume that whatever string follows the "@" is a dblink. Just look for Token.AT followed by Token.STRING; if we find this sequence immediately following the table name, then we record that as the db link.

Is there an issue with doing that that I've missed? Generally the philosophy of the parser is to not interpret anything unless we absolutely have to. If there's some other `Token.AT, Token.STRING` sequence that _doesn't_ relate to DB links, that would be an issue. 

Second: rather than looking for those tokens in this switch, look for the sequence after we break out of the loop, after finding the table name. Similar in appendIdentifiers. This would probably mean making a few small changes: break the loop rather than returning immediately in the default case, and widen the scope of `t`.